index.rst

```diff
@@ -10,16 +10,200 @@
+First, we will try to transfer the outputs from failed (i.e., incomplete) pipelines, as these may help in diagnosing the problem.
+Such a strategy is safe so long as no code assumes that the existence of one dataset implies the existence of another.
+Second, the central repository sync itself may fail, leaving an undefined subset of datasets transferred.
+Again, the immediate risk is that something might assume the repository contains a self-consistent set of outputs; in the longer term, the datasets can be regenerated through daytime processing.
+
+
+.. _bad-data:
+
+Corrupted Pipeline Outputs
+==========================
+
+It's possible that some processing errors will allow the pipeline to run to completion while producing large numbers of invalid sources.
+Such sources will clutter the alert stream with false positives, and may confuse source association on later visits.
+
+The Prompt Processing framework does not itself have any way to detect nonsense output.
+However, the Alert Production team is incorporating "circuit breaker" checks into the pipeline; see `DM-35645`_ and its follow-up issues.
+These checks will escalate suspicious outputs into pipeline failures, which can be handled as described above.
+As of May 2023, the proposed checks focus on poor-quality raw inputs; there are no checks specifically guarding DiaSouce detection or the APDB.
+
+.. _DM-35645: https://jira.lsstcorp.org/browse/DM-35645
+
+If invalid sources are reported through the alert stream, a way to retract alerts will be useful.
+Such a design is described in `DMTN-259`_.
+It's out of scope for this note, since retraction will require human intervention and cannot be done at processing time.
+
+.. _DMTN-259: https://dmtn-259.lsst.io/
+
+.. _timeout:
+
+Pipeline Timeouts
+=================
+
+Another risk is that, under some circumstances, the pipeline code may fail to terminate.
+This risk is mitigated by the scalable processing framework; taking a single worker out of action will not interfere with the processing of other images or with overall system performance.
+
+The obvious way to handle stuck pipelines is to impose a timeout on either the job or the worker.
+The former carries the risk that the worker will be left in an inconsistent state, corrupting future processing, while the latter loses all intermediate data and adds the overhead of starting a new worker and preparing its local repository.
+In either case, the possibility of a timeout needs to be accounted for when solving the problems discussed in :ref:`association` and :ref:`consistency`, since a timeout can interrupt processing at *any* time, including during I/O.
+
+It might be possible to impose a timeout at the task level rather than the worker level, to better distinguish between a slow job and a stuck one.
+However, no current framework allows tasks to be timed in real time; for example, timing metrics are only available after a task has completed.
```

Comment:
There are reasons for DRP to impose timeouts as well; it's possible that these might be pushed to the task level rather than the workflow job level in a future iteration of the middleware.